go/sentry/client/client.go

```diff
@@ -45,9 +45,11 @@ func (c *Client) createConnection() error {
 	certPool := x509.NewCertPool()
 	certPool.AddCert(c.sentryCert)
 	creds := credentials.NewTLS(&tls.Config{
-		Certificates: []tls.Certificate{*c.nodeIdentity.TLSCertificate},
-		RootCAs:      certPool,
-		ServerName:   identity.CommonName,
+		RootCAs:    certPool,
+		ServerName: identity.CommonName,
+		GetClientCertificate: func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+			return c.nodeIdentity.GetTLSCertificate(), nil
```

Comment:
This will probably not work as this certificate can be rotated, the control connection should use its own long-term certs?

Reply:
Yes, I'll add that as part of this PR :)